web/iot/iot.js

```diff
@@ -0,0 +1,231 @@
+
+/**
+* - With PubNub you can implement any real-time solution possible.
+* - In this code base, we will explore how you would use PubNub to communicate over PubNub Signals with IoT Devices.
+* - This file deals with all the sending and receiving of information to and from the devices
+*/
+
+//  Connection to the PubNub API
+var pubnub = null
+// List of IoT Devices and their readings
+var iotDevices = null
+// Current Values for alarm settings
+var alarmSettings = null;
+// Current Values for temperature settings
+var valueSettings = null;
+
+// Called on page load
+async function initialize () {
+
+  // Declarations
+  iotDevices = {}
+  alarmSettings = {};
+  valueSettings = {}
+
+  //  PubNub object - connection with the PubNub infrastructure
+  pubnub = createPubNubObject();
+
+  activatePubNubListener(); // Listen to channels device.* for any updates
+
+  //  Subscribing to all possible channels we will want to know about.
+  //  Need to know about device.* channel as it will send us updates of sensor readings
+  //  Using the recommended naming convention:
+  //  Public.<name> for public groups
+  //  Private.<name> for private groups
+  //  DM.A&B for direct messages between two users
+  pubnub.subscribe({channels: ["device.*", `Private.${pubnub.getUserId()}-iot`], withPresence: true}); // Subscribe
```

Comment:
I don't think you need to subscribe to `Private.${pubnub.getUserId()}-iot` since you will not be receiving data on that channel, only publishing it?  Also, do you use presence?